runners/google-cloud-dataflow-java/examples-streaming/build.gradle

```diff
@@ -81,8 +104,38 @@ task appliancePreCommit(type: Test) {
   systemProperty "beamTestPipelineOptions", JsonOutput.toJson(preCommitBeamTestPipelineOptions)
 }
 
+task appliancePreCommitJava11(type: Test) {
+  dependsOn ":runners:google-cloud-dataflow-java:worker:legacy-worker:shadowJar"
+  def dataflowWorkerJar = project.findProperty('dataflowWorkerJar') ?: project(":runners:google-cloud-dataflow-java:worker:legacy-worker").shadowJar.archivePath
+
+  // Set workerHarnessContainerImage to empty to make Dataflow pick up the non-versioned container
+  // image, which handles a staged worker jar.
+  def preCommitBeamTestPipelineOptions = [
+          "--project=${gcpProject}",
+          "--tempRoot=${gcsTempRoot}",
+          "--runner=TestDataflowRunner",
+          "--dataflowWorkerJar=${dataflowWorkerJar}",
+          "--workerHarnessContainerImage=",
+          "--streaming=true",
+  ]
+  testClassesDirs = files(project(":examples:java").sourceSets.test.output.classesDirs)
+  include "**/WordCountIT.class"
+  // Don't include WindowedWordCountIT, since it is already included in
+  // :runners:google-cloud-dataflow-java:examples:preCommit and runs
+  // identically (ignores the --streaming flag).
+  forkEvery 1
+  maxParallelForks 4
+  systemProperty "beamTestPipelineOptions", JsonOutput.toJson(preCommitBeamTestPipelineOptions)
+  systemProperty "java.specification.version", "11"
```

Comment:
I'm working on a similar task and I just found out that the `java.specification.version` property is supplied by the JVM.
This makes all the duplicated tasks unnecessary - everything works by passing proper Java home to Gradle.

Please remove the duplicates.

Reply:
Thanks for input!